src/plugins/serverEvents.js

```diff
@@ -0,0 +1,43 @@
+import { SSE, WS } from '../server_events'
+
+// Available providers; priority determined by order
+// @note move to config?
+const providers = [
+  { Client: WS },
+  { Client: SSE },
+]
+
+export default {
+  install (Vue, { eventbus }) {
+    Vue.prototype.$ServerEvents = {
+      client: null,
+
+      connect (api, jwt) {
+        this.determineClient(api, jwt)
+          .then(({ Client }) => {
+            this.client = (new Client({ api, jwt, eventbus })).connect()
+          })
+          .catch((e) => {
+            eventbus.$emit('$ServerEvents.noClient')
+          })
+      },
+
+      disconnect () {
+        if (this.client) {
+          this.client.disconnect()
+        }
+      },
+
+      async determineClient (api, jwt) {
+        for (let { Client } of providers) {
```

Comment:
This is overdesigned. No need to support multiple client types.
As I see this, we need one client for event handling (may that be via WS for now and one for SSE when we switch) + one client for simple REST short/long polling as a fallback.